scripts/testpop/statter.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/cactus/go-statsd-client/statsd"
+)
+
+// WriteCloserToSender is an adapter from io.WriteCloser to statsd.Sender
+type WriteCloserToSender struct {
+	bufWriter *bufio.Writer
+	closer    io.Closer
+}
+
+// Send implements the statsd.Sender interface
+func (w *WriteCloserToSender) Send(data []byte) (int, error) {
+	return w.bufWriter.Write(data)
+}
+
+// Close implements the statsd.Sender interface
+func (w *WriteCloserToSender) Close() error {
+	err := w.bufWriter.Flush()
+	w.closer.Close()
+	return err
+}
+
+// NewWriteCloserToSender returns a new WriteCloserToSender adapter
+func NewWriteCloserToSender(wc io.WriteCloser) statsd.Sender {
```

Comment:
It think it's more useful to return a WriterCloserToSender and comment that it implements a statsd.Sender.
Same that os.Open returns a *File which implements the reader interface